framework/src/modules/nft/endpoint.ts

```diff
@@ -143,33 +148,30 @@ export class NFTEndpoint extends BaseEndpoint {
 
 		validator.validate<{ chainID: string }>(getSupportedCollectionIDsRequestSchema, params);
 
+		const { ownChainID } = this._config;
 		const chainID = Buffer.from(params.chainID, 'hex');
-		const nftID = Buffer.concat([chainID, Buffer.alloc(12)]);
+		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
 
-		const isNFTSupported = await this._nftMethod.isNFTSupported(
-			context.getImmutableMethodContext(),
-			nftID,
-		);
+		const areAllNFTsSupported = await supportedNFTsStore.has(context, ALL_SUPPORTED_NFTS_KEY);
+		if (areAllNFTsSupported) {
+			return { collectionIDs: [] };
+		}
 
-		if (!isNFTSupported) {
-			return undefined;
+		const isSupported = await supportedNFTsStore.has(context, chainID);
+		if (!isSupported) {
+			return chainID.equals(ownChainID) ? { collectionIDs: [] } : undefined;
 		}
 
-		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
 		const supportedNFTsData = await supportedNFTsStore.get(
 			context.getImmutableMethodContext(),
 			chainID,
 		);
 
-		if (supportedNFTsData.supportedCollectionIDArray.length) {
-			return {
-				collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
-					collection.collectionID.toString('hex'),
-				),
-			};
-		}
-
-		return undefined;
+		return {
+			collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
+				collection.collectionID.toString('hex'),
+			),
+		};
```

Comment:
Here you will get `[]` if `supportedCollectionIDArray` is empty, but it should be `['*']` since no entries means all collections are supported. So I suggest something like this before `return`:
```
if (supportedNFTsData.supportedCollectionIDArray.length === 0) {
    return { collectionIDs: ['*'] };
}
```

Reply:
It seems to me that for native NFT collections it's much different than with native tokens.

Reply:
I missed the last part about Native collections....lemme address this.